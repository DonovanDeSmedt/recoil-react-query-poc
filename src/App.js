import React from 'react';
import { ErrorBoundary } from './modules/ui';
import { ThemeSelector } from './modules/ui/themeSelector';
import { TodoModule } from './modules/todo';

const Apps = () => {
  // Adding an error boundary on the highest level,
  // to catch unexpected JS errors.
  // Because this is the highest level, we put another
  return (
    <ErrorBoundary
      message={
        'The application crashed. Please refresh your browser to proceed.'
      }>
      <div
        style={{
          display: 'flex',
          flexDirection: 'row',
          flex: 1,
          justifyContent: 'space-between',
        }}>
        <TodoModule />
        <ThemeSelector />
      </div>
    </ErrorBoundary>
  );
};

export default Apps;
