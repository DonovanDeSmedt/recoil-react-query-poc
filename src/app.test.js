import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';

test('App renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('App renders welcome message', () => {
  const app = renderer.create(<App />);
  const tree = app.toJSON();
  expect(tree).toMatchSnapshot();
});
