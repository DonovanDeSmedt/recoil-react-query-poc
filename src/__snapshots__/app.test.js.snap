// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders welcome message 1`] = `
Array [
  <p>
    Import your first component here (app.js)
  </p>,
  <p>
    Happy coding!
  </p>,
]
`;
